### DISCUSSION TITLE
Using docker to build a farmer, a wallet will be created automatically after each restart.

### DISCUSSION THREAD

**Discussion by coinsccg** [Chia-Network/chia-blockchain#3255] [Support]

**Comment by KickCellarDoor** [comment]
Check -e keys=".." option for Docker, make sure it is not "generate"

**Comment by coinsccg** [comment]
> Check -e keys=".." option for Docker, make sure it is not "generate"

Does this key refer to wallet public keys？

**Comment by KickCellarDoor** [comment]
@qian-fei It should be the path of the key file in the docker container.

**Comment by KickCellarDoor** [comment]
Check this [instruction](https://github.com/orgs/Chia-Network/packages/container/package/chia) .

**Comment by 007** [comment]
I napped a drive for the config, so I should have persistent storage as if I'm running it natively. I still get this key duplication.

I didn't set a key file on first startup, why isn't it using some default well-known path for the keyfile instead of creating a new one every run?

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
